### ISSUE TITLE
Support MQTT messages that have observation names that differ from the WeeWX names

### ISSUE THREAD

**Issue by bellrichm** [bellrichm/weewx-jas#33]

**Comment by gary-hammer** [comment]
I'll make this easy on you.
This instance of WeeWX is fed by your MQTTSubscribe.

**Comment by bellrichm** [comment]
Sounds kinda’ like what I do. Publish MQTT from my ‘production’ instance and use MQTTSubscribeDriver on my development instances. WeeWX-jas is also subscribing, so you are probably using [weewx-mqtt](https://github.com/matthewwall/weewx-mqtt) to publish?
Net, I’d really like to know what the actual MQTT data looks like.
Again, thanks for your help.

**Comment by gary-hammer** [comment]
Davis Vantage Pro2 Plus via logger to WeeWX
weewx-mqtt publishes to my local mosquitto broker
wee-extension --list show
mqtt              0.23      Upload weather data to MQTT server.

I'm attaching 3 weather/loop messages as mqtt-msg.txt
Let me know if there is anything else you'd like.
[mqtt-msg.txt](https://github.com/bellrichm/weewx-jas/files/9289100/mqtt-msg.txt)

**Comment by gary-hammer** [comment]
Just thought about this.
How is MQTT enabled for the skin?
It would have to use websockets I'd guess or does it?
Here's my config, minus the user info, it is using websockets.

[[[[mqtt]]]]
                enable = true
                
                host = weather.ghammer.net
                port = 9001
                
                useSSL = true
                
                username = xxxxxxx
                password = xxxxxxx
                
                topic = weather/loop

**Comment by bellrichm** [comment]
By default `weewx-mqtt` appends the observation’s units, `outTemp_F`, `windSpeed_mph`, `rainRate_inch_per_hour`, etc. You can change this via, [append_units_label](https://github.com/matthewwall/weewx-mqtt/blob/78b19fa5ffc21ebcb0fb4fdc4cbe233ad3d5b929/bin/user/mqtt.py#L227)
I have to think a bit about how I want to handle the mapping and set up an environment to develop it in.

**Comment by bellrichm** [comment]
It would probably break them. That is why I am looking at being able to map names. Hopefully it won’t be too hard.

**Comment by gary-hammer** [comment]
That's what I thought.
I'll wait for your go ahead.

**Comment by bellrichm** [comment]
Just verifying. I’m assuming that your MQTTSubscribeDriver maps incoming names with units appended to names without the units…
Thanks

**Comment by gary-hammer** [comment]
It is the output/messages straight from the MQTT server that the production
WeeWX sends to.

**Comment by bellrichm** [comment]
To map incoming MQTT names to WeeWX names, a new `[[[[[fields]]]]]` section will be added under the `[[[[mqtt]]]]` section. The following was used as the code was developed.
```
                [[[[[fields]]]]]
                    [[[[[[barometer_inHg]]]]]]
                        name = barometer
                    [[[[[[dewpoint_F]]]]]]
                        name = dewpoint
                    #[[[[[[ET]]]]]] not in MQTT data
                    [[[[[[heatindex_F]]]]]]
                        name = heatindex
                    [[[[[[inTemp_F]]]]]]
                        name = inTemp
                    #[[[[[[outHumidity]]]]]] no units
                    [[[[[[outTemp_F]]]]]]
                        name = outTemp
                    [[[[[[radiation_Wpm2]]]]]]
                        name = radiation
                    [[[[[[rain_in]]]]]]
                        name = rain
                    [[[[[[rainRate_inch_per_hour]]]]]]
                        name = rainRate
                    #[[[[[[sun]]]]]] not in MQTT data
                    #[[[[[[UV]]]]]] no units
                    [[[[[[windchill_F]]]]]]
                        name = windchill
                    [[[[[[windSpeed_mph]]]]]]
                        name = windSpeed
```
Officially documentation to come.

**Comment by bellrichm** [comment]
Implemented in [v0.2.2-rc03](https://github.com/bellrichm/weewx-jas/releases/tag/v0.2.2-rc03)
